src/main/java/PersonalBudget/business/expense/domain/service/ExpenseService.java

```diff
@@ -0,0 +1,79 @@
+package PersonalBudget.business.expense.domain.service;
+
+import PersonalBudget.business.expense.domain.mapper.ExpenseMapper;
+import PersonalBudget.business.expense.domain.model.ExpenseCategoryEntity;
+import PersonalBudget.business.expense.domain.model.ExpenseDefaultCategory;
+import PersonalBudget.business.expense.domain.model.ExpenseDefaultPaymentMethod;
+import PersonalBudget.business.expense.domain.model.ExpenseEntity;
+import PersonalBudget.business.expense.domain.model.ExpensePaymentMethodEntity;
+import PersonalBudget.business.expense.domain.repository.ExpenseCategoryRepository;
+import PersonalBudget.business.expense.domain.repository.ExpensePaymentMethodRepository;
+import PersonalBudget.business.expense.domain.repository.ExpenseRepository;
+import PersonalBudget.business.expense.dto.ExpenseCategoryDTO;
+import PersonalBudget.business.expense.dto.ExpenseDTO;
+import PersonalBudget.business.expense.dto.ExpensePaymentMethodDTO;
+import PersonalBudget.business.user.domain.UserFacade;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+@RequiredArgsConstructor
+@Service
+public class ExpenseService {
+
+    private final ExpenseCategoryRepository expenseCategoryRepository;
+    private final ExpensePaymentMethodRepository paymentMethodRepository;
+    private final ExpenseRepository expenseRepository;
+    private final ExpenseMapper expenseMapper;
+    private final UserFacade userFacade;
+
+    public List<ExpenseCategoryDTO> getUserExpenseCategories() {
+        Long loggedInUserId = userFacade.fetchLoggedInUserId();
+        return expenseCategoryRepository.findAllExpenseCategoryByUserId(loggedInUserId);
+    }
+
+    public List<ExpensePaymentMethodDTO> getUserPaymentMethods() {
+        Long loggedInUserId = userFacade.fetchLoggedInUserId();
+        return paymentMethodRepository.findAllPaymentMethodName(loggedInUserId);
+    }
+
+    public void addExpense(ExpenseDTO expenseDTO) {
+        Long userId = userFacade.fetchLoggedInUserId();
+        Long userExpenseCategoryId = expenseDTO.expenseCategoryId();
+        Long userPaymentMethodId = expenseDTO.paymentMethodId();
+        ExpenseEntity expenseEntity = expenseMapper.mapExpenseDTOToExpenseEntity(expenseDTO, userId, userExpenseCategoryId, userPaymentMethodId);
```

Comment:
You pass DTO to the mapper, why do you pass userExpenseCategoryId and userExpenseCategoryId separately?